### PR TITLE
Code Refactoring

### PR DESCRIPTION
**Pull request by PhoneDroid**
#2 

https://phonedroid.github.io/sorting_algos/

Also configured GitHub Actions for static page content:
- Uses the /Source/ folder for the page
- Only rebuilds the page if any of the files in the /Source/ folder change

In a followup PR I would like to address bootstrap as well,
99+% percent of it isn't used ( checked the coverage ).

I would replace it with tailwindcss, which generates the required classes
which on one hand removes  all the unused parts and on the other is
freely adjustable.

### CODE REVIEW

**Comment by PhoneDroid** [comment]
@daweedkob You might want to take a look, I got most of it untangled.

Btw, every commit was usable and so it the latest.

**Comment by PhoneDroid** [comment]
In this reworked version, you can also adjust the speed while it's running as it doesn't precalculate the timings.

**Comment by PhoneDroid** [comment]
Wrote a small [README](https://github.com/PhoneDroid/sorting_algos) and linked all the algorithms

File: Source/index.html [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                            class='fas fa-sun' 
+                            title = 'Light Mode'
+                        ></i>
+                    </a>
+                    
+                    <a 
+                        class = 'navbar-brand about'
+                        href = 'https://www.linkedin.com/in/the-david-kobalia/'
+                    >Made By David Kobalia</a>
+
```
Comment (daweedkob) [inline Source/index.html:53]: On the phone version of the website, you will not be able to select algorithms or specify size/speed. Consider adding a button.

```
<button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#nav-dropbar" aria-controls="nav-dropbar" aria-expanded="false" aria-label="Toggle navigation">
                        <span class="navbar-toggler-icon"></span>
                    </button>
```

Comment (PhoneDroid) [reply Source/index.html:53]: Why not just remove the 'collapse' class?

**Comment by daweedkob** [comment]
First, of @PhoneDroid  really good work, I like what you did with the project's architecture.

Though I have found small bugs, when those will be fixed I will take a last look and merge the changes.

* On Safari browser bars do not show up (plus on the desktop version light mode night mode switch does not work)
image

* On the phone version of the website you can't specify the algorithm nor size and speed. (Added in the comment)

How it is right now:
image

How it can be fixed:
image

Collapsed:
image

**Comment by PhoneDroid** [comment]
> On Safari browser bars do not show up (plus on the desktop version light mode night mode switch does not work)

Is there anything in the console?

File: Source/index.html (hunk 1 above)
Comment (daweedkob) [reply Source/index.html:53]: If you just remove `collapse` class with the current implementation, in the small resolution it will be a mess.

**Comment by daweedkob** [comment]
> > On Safari browser bars do not show up (plus on the desktop version light mode night mode switch does not work)
> 
> Is there anything in the console?

image

**Comment by PhoneDroid** [comment]
Lol, safafi doesn't support importmaps?

**Comment by PhoneDroid** [comment]
Not sure how you can get that endsWith error considering that all stylesheets should have a href ..
I can use `?.` I guess, but I wonder how it got to that ..

**Comment by PhoneDroid** [comment]
They should all support it https://developer.mozilla.org/en-US/docs/Web/API/StyleSheet

**Comment by PhoneDroid** [comment]
It can only be null if it was an inline stylesheet .. https://developer.mozilla.org/en-US/docs/Web/API/StyleSheet/href

**Comment by PhoneDroid** [comment]
This shim might work for safari: https://www.bram.us/2021/09/13/support-import-maps-in-any-browser-with-es-module-shims/

**Comment by daweedkob** [comment]
> Not sure how you can get that endsWith error considering that all stylesheets should have a href .. I can use `?.` I guess, but I wonder how it got to that ..

Log of `sheets`:
image

**Comment by PhoneDroid** [comment]
Oh, do you use any extensions perhaps?
If so, they might be injecting stylesheets which would be inline and thus `href = null`

**Comment by PhoneDroid** [comment]
Should I add the shim, can't test it ( don't have safari ), you could tell me if it worked I guess ..

**Comment by daweedkob** [comment]
> Oh, do you use any extensions perhaps? If so, they might be injecting stylesheets which would be inline and thus `href = null`

Yes, you are right extensions are the cause of it.

**Comment by daweedkob** [comment]
> Should I add the shim, can't test it ( don't have safari ), you could tell me if it worked I guess ..

Yup, added shim works like a charm.

**Comment by PhoneDroid** [comment]
Guess I'll add it then..

**Comment by daweedkob** [comment]
I will add it, no worries. Also, I would like to move the collapse button up a bit.

**Comment by PhoneDroid** [comment]
> I will add it, no worries. 

So should I revert it .. ?

**Review by daweedkob — COMMENTED** [review]
Moved up the collapse button

**Comment by PhoneDroid** [comment]
Btw, since you agreed to doing it with branches,
how about you create a new branch from your
current, call it `original` and accept my request
on the main branch.

Afterwards you can link to the original branch.

**Comment by daweedkob** [comment]
I like how it is implemented rn, so I will merge it into the main. Nicely done.

**Pull request merged**